tests/test_pyglider.py

```diff
@@ -57,6 +58,98 @@ def test_example_seaexplorer_metadata():
+output_interp = xr.open_dataset(outname_interp)
+
+@pytest.mark.parametrize("var", variables)
+def test_example_seaexplorer_interp_nrt(var):
+    assert output_interp[var].attrs == test_data[var].attrs
+    if var not in ['time']:
+        np.testing.assert_allclose(output_interp[var].values, test_data[var].values, rtol=1e-5)
+    else:
+        dt0 = output_interp[var].values - np.datetime64('2000-01-01')
+        dt1 = test_data[var].values - np.datetime64('2000-01-01')
+        assert np.allclose(
+            np.array(dt0, dtype='float64'),
+            np.array(dt1, dtype='float64'))
+
+
+# Test raw (full resolution) seaexplorer data.
+
+rawdir = str(example_dir / 'example-seaexplorer-raw/delayed_raw/') + '/'
+rawncdir = str(example_dir / 'example-seaexplorer-raw/delayed_rawnc/') + '/'
+deploymentyaml_raw = str(example_dir / 'example-seaexplorer-raw/deployment.yml')
+l0tsdir = str(example_dir / 'example-seaexplorer-raw/L0-timeseries-test/') + '/'
+seaexplorer.raw_to_rawnc(rawdir, rawncdir, deploymentyaml_raw)
+seaexplorer.merge_parquet(rawncdir, rawncdir, deploymentyaml_raw, kind='raw')
+outname_raw = seaexplorer.raw_to_L0timeseries(rawncdir, l0tsdir,
+                                          deploymentyaml_raw, kind='raw')
+output_raw = xr.open_dataset(outname_raw)
+# Open test data file
+test_data_raw = xr.open_dataset(
+    library_dir /
+    'tests/results/example-seaexplorer-raw/L0-timeseries/dfo-bb046-20200908.nc')
+
+@pytest.mark.parametrize("var", variables)
+def test_example_seaexplorer_raw(var):
+    # Test that each variable and its coordinates match
+    assert output_raw[var].attrs == test_data_raw[var].attrs
+    if var not in ['time']:
+        np.testing.assert_allclose(output_raw[var].values, test_data_raw[var].values, rtol=1e-5)
+    else:
+        dt0 = output_raw[var].values - np.datetime64('2000-01-01')
+        dt1 = test_data_raw[var].values - np.datetime64('2000-01-01')
+        assert np.allclose(
+            np.array(dt0, dtype='float64'),
+            np.array(dt1, dtype='float64'))
+
+
+def test_example_seaexplorer_metadata_raw():
+    # Test that attributes match. Have to remove creation and issue dates first
+    output_raw.attrs.pop('date_created')
+    output_raw.attrs.pop('date_issued')
+    test_data_raw.attrs.pop('date_created')
+    test_data_raw.attrs.pop('date_issued')
+    assert output_raw.attrs == test_data_raw.attrs
+
+
+# Test nan interpolation on raw data.
+
+# Test that interpolation over nans in raw data results in a greater or equal number of non-nan values
+with open(deploymentyaml_raw) as fin:
+    deployment_raw = yaml.safe_load(fin)
+interp_yaml = str(example_dir / 'example-seaexplorer-raw/deploymentRealtimeInterp.yml')
+deployment_raw['netcdf_variables']["interpolate"] = True
+with open(interp_yaml, "w") as fout:
+    yaml.dump(deployment_raw, fout)
+l0tsdir_interp_raw = str(example_dir / 'example-seaexplorer-raw/L0-timeseries-test-interp/') + '/'
```

Comment:
I think this will overwrite the results of the test for the subsampled interpolated data above, which isn't a major issue, but it would be nice to have a separate directory for the delayed-mode interpolated data in case we want to manually look at the resulting files.

Reply:
It uses the same raw directory `rawnc` as the run on raw data above, as the interpolation step comes after the `raw_to_raw_nc` and `merge_parquet` functions. This prevents unnecessary re-runs of the code. It writes results out to a new directory `l0tsdir_interp_raw` which is different to `l0ts_dir` and `l0tsdir_interp` so that they can be compared with the results obtained not using the interpolation functionality. 

I appreciate the naming of these is quite confusing. I think all the examples could benefit from a cleanup/refactor in a future PR